File: FrontEnd/src/app/Class/albums.ts
import { Bands } from './bands'
import { Songs } from './songs'

export class Albums{

	private id: number;
	private albumName: String;
	private band: String;
	private songs: String;
	private lengthOfAlbum: number;
	private averageRateOfAlbum: number;

	constructor(obj?: any){
	
		this.id = (obj && obj.id) || 0;
		this.albumName = (obj && obj.albumName) || '';
		this.band = (obj && obj.band) || '';
		this.songs = (obj && obj.songs) || '';
		this.lengthOfAlbum = (obj && obj.lengthOfAlbum) || 0;
		this.averageRateOfAlbum = (obj && obj.averageRateOfAlbum) || 0;

	}

	getId(): number{
		return this.id;
	}
	setId(id: number){
		this.id = id;
	}
	
	getAlbumName(): String{
		return this.albumName;
	}
	setAlbumName(albumName: String){
		this.albumName = albumName;
	}
	
	getBand(): String{
		return this.band;
	}
	setBand(band: String){
		this.band = band;
	}
	
	getSongs(): String{
		return this.songs;
	}
	setSongs(songs: String){
		this.songs = songs;
	}
	
	getLengthOfAlbum(): number{
		return this.lengthOfAlbum;
	}
	setLengthOfAlbum(lengthOfAlbum: number){
		this.lengthOfAlbum = lengthOfAlbum;
	}

	getAverageRateOfAlbum(): number{
		return this.averageRateOfAlbum;
	}
	setAverageRateOfAlbum(averageRateOfAlbum: number){
		this.averageRateOfAlbum = averageRateOfAlbum;
	}

}

File: Backend/src/main/java/Spring/Entities/Albums.java
package Spring.Entities;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "Albums", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"id"})
})
public class Albums {

    @Id
    @GeneratedValue
    @Column(name = "id")
    private int id;

    @Column(name = "Album_Name")
    private String albumName;

    @Column(name = "Band")
    private String band;

    @Column(name = "Songs")
    private String songs;

    @Column(name = "Lenght_Of_Album")
    private int lengthOfAlbum;

    @Column(name = "Average_Rate_Of_Album")
    private float averageRateOfAlbum;

    //Constructors

    public Albums(){
    }

    public Albums(String albumName, int lengthOfAlbum, String band, float averageRateOfAlbum) {
        this.albumName = albumName;
        this.lengthOfAlbum = lengthOfAlbum;
        this.band = band;
        this.averageRateOfAlbum = averageRateOfAlbum;
    }

    //Getters and Setters


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getAlbumName() {
        return albumName;
    }

    public void setAlbumName(String albumName) {
        this.albumName = albumName;
    }

    public int getLengthOfAlbum() {
        return lengthOfAlbum;
    }

    public void setLengthOfAlbum(int lengthOfAlbum) {
        this.lengthOfAlbum = lengthOfAlbum;
    }

    public float getAverangeRateOfAlbum() {
        return averageRateOfAlbum;
    }

    public void setAverangeRateOfAlbum(float averageRateOfAlbum) {
        this.averageRateOfAlbum = averageRateOfAlbum;
    }

    public void setSongs(String songs) {
        this.songs = songs;
    }
    public String getSongs(){
        return songs;
    }

    public String getBand() {
        return band;
    }

    public void setBand(String band) {
        this.band = band;
    }
}
